generator/src/test/java/com/scottlogic/deg/generator/cucumber/AfterOrAt.feature

```diff
@@ -0,0 +1,177 @@
+       And foo is in set:
+       | 2016-01-01T12:00:00.000 |
+       | 2017-01-01T12:00:00.000 |
+       | 2018-01-01T12:00:00.000 |
+       | 2019-01-01T12:00:00.000 |
+       | 2019-01-01T12:01:05.000 |
+       | 2020-01-01T12:01:00.000 |
+     Then the following data should be generated:
+       | foo                     |
+       | 2018-01-01T12:00:00.000 |
+       | 2019-01-01T12:00:00.000 |
+       | 2019-01-01T12:01:05.000 |
+       | 2020-01-01T12:01:00.000 |
+
+Scenario: Running afterOrAt request that includes temporal field with date (YYYY-MM-DD) values that has invalid date should fail
+     Given foo is after or at 2019-15-32T00:00:00.000
+     Then I am presented with an error message
+       And no data is created
+
+Scenario: Running afterOrAt request that includes temporal field with date and time (YYYY-MM-DDTHH:MM:SS) values that has invalid time should fail
+     Given foo is after or at 2018-10-01T25:25:05.000
+     Then I am presented with an error message
+       And no data is created
+
+Scenario: Running afterOrAt request that includes temporal field with date and time (YYYY-MM-DDTHH:MM:SS) values that has invalid year should fail
+     Given foo is after or at 0000-01-10T00:00:00.000
+     Then I am presented with an error message
+       And no data is created
+
+Scenario: Running beforeOrAt request that includes temporal field with date and time (YYYY-MM-DDTHH:MM:SS) values that has invalid format should fail
+     Given foo is after or at "2018-Jan-31stT00:00:00.000"
+     Then I am presented with an error message
+       And no data is created
+
+
+Scenario: Running afterOrAt request that includes temporal field with date and time (YYYY-MM-DDTHH:MM:SS) values that has leap year
+     Given foo is after or at 2019-01-01T00:00:00.000
+       And foo is in set:
+       | 2017-01-01T00:00:00.000 |
+       | 2018-01-01T00:00:00.000 |
+       | 2019-01-01T00:00:00.000 |
+       | 2020-01-01T00:00:00.000 |
+       | 2021-01-01T00:00:00.000 |
+     Then the following data should be generated:
+       | foo                     |
+       | 2019-01-01T00:00:00.000 |
+       | 2020-01-01T00:00:00.000 |
+       | 2021-01-01T00:00:00.000 |
+
+
+Scenario: Running afterOrAt request against a non-contradicting afterOrAt constraint should be successful
+     Given foo is after or at 2019-01-01T00:00:00.000
+       And foo is after or at 2020-01-01T00:00:00.000
+       And foo is in set:
+       | 2017-01-01T00:00:00.000 |
+       | 2018-01-01T00:00:00.000 |
+       | 2019-01-01T00:00:00.000 |
+       | 2020-01-01T00:00:00.000 |
+       | 2021-01-01T00:00:00.000 |
+       | 2022-01-01T00:00:00.000 |
+     Then the following data should be generated:
+       | foo                     |
+       | 2020-01-01T00:00:00.000 |
+       | 2021-01-01T00:00:00.000 |
+       | 2022-01-01T00:00:00.000 |
+
+Scenario: Running afterOrAt request against a non contradicting not afterOrAtOrAt should be successful
```

Comment:
This scenario describes a "not afterOrAtOrAt" but doesn't contain a step that says `anything but after or at...`

Note that it should also be "not afterOrAt"

This test is a duplicate of the prior scenario

Reply:
Yes, it was a duplicate scenario. I removed it .